### DISCUSSION TITLE
error: no member named 'p' in 'AIR::(lambda ...)

### DISCUSSION THREAD

**Discussion by OfekShochat** [foonathan/lexy#158] [Q&A] [answered]
Hello!
For some reason when I write something like:
```struct AIR {
    static constexpr auto max_recursion_depth = 19;

    static constexpr auto whitespace = dsl::ascii::blank / dsl::ascii::newline;

    static constexpr auto rule = [] {
        return dsl::p<InstructionRule> + dsl::eof;
    };

    static constexpr auto value = lexy::forward<Instruction>;
};
```
compilation errors out with  `error: no member named 'p' in 'AIR::(lambda at parser/parser.h:63:34)'` (at https://github.com/foonathan/lexy/blob/main/include/lexy/dsl/base.hpp#L100). I didn't see any inheritance done in the examples, and I couldn't find anything about it online. I might have an idea of what it is related to, but no idea how to fix it.
Could you help?
Thanks

**Comment by Life4gal** [comment]
I think you missed [this](https://github.com/foonathan/lexy/discussions/97#discussioncomment-4062767). :)

**Comment by Life4gal** [comment, marked as answer]
```cpp
static constexpr auto rule = [] {
        return dsl::p<InstructionRule> + dsl::eof;
    };
```
|
|
v
```cpp
static constexpr auto rule = [] {
        return dsl::p<InstructionRule> + dsl::eof;
    }();
```

**Reply by OfekShochat** [reply]
Oh well
